lib/new_relic/agent/instrumentation/elasticsearch/instrumentation.rb

```diff
@@ -0,0 +1,55 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+require_relative '../../datastores/nosql_obfuscator'
+
+module NewRelic::Agent::Instrumentation
+  module Elasticsearch
+    PRODUCT_NAME = 'Elasticsearch'
+    OPERATION = 'query'
+
+    def perform_request_with_tracing(method, path, params = {}, body = nil, headers = nil)
```

Comment:
Do we anticipate Elasticsearch ever modifying the params hash downstream? If not, we could use `NewRelic::EMPTY_HASH` here on the `def` line as an optimisation.

Reply:
Hmm, I don't think it would modify it later, but I am hesitant to do that. Usually I try to match the original methods args, so I do worry that using a frozen could potentially cause something weird to happen later. I'll look into it more tomorrow

Reply:
That's a good call. If the original method was instantiating a new hash, then we are not introducing any extra allocations by being the one to spawn it and hand it off downstream. And if we stick with an unfrozen empty hash as the original code uses, we're guaranteed not to break anything. I like that - let's just keep this as is.